tests/west/project/test_project.py

```diff
@@ -0,0 +1,200 @@
+import argparse
+import os
+import shlex
+import tempfile
+
+import pytest
+
+import west.cmd.project
+
+
+MANIFEST_PATH = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), 'manifest.yml')
+)
+
+# Where the projects are cloned to
+NET_TOOLS_PATH = 'net-tools'
+KCONFIGLIB_PATH = 'sub/kconfiglib'
+
+COMMAND_OBJECTS = (
+    west.cmd.project.ListProjects(),
+    west.cmd.project.Fetch(),
+    west.cmd.project.Pull(),
+    west.cmd.project.Rebase(),
+    west.cmd.project.Branch(),
+    west.cmd.project.Checkout(),
+    west.cmd.project.Diff(),
+    west.cmd.project.Status(),
+)
+
+
+def cmd(cmd):
+    cmd += ' -m ' + MANIFEST_PATH
+
+    # cmd() takes the command as a string, which is less clunky to work with.
+    # Split it according to shell rules.
+    split_cmd = shlex.split(cmd)
+    command_name = split_cmd[0]
+
+    for command_object in COMMAND_OBJECTS:
+        # Find the WestCommand object that implements the command
+        if command_object.name == command_name:
+            # Use it to parse the arguments
+            parser = argparse.ArgumentParser()
+            command_object.do_add_parser(parser.add_subparsers())
+
+            # Pass the parsed arguments and unknown arguments to run it
+            command_object.do_run(*parser.parse_known_args(shlex.split(cmd)))
+            break
+    else:
+        assert False, "unknown command " + command_name
+
+
+# Example taken from https://docs.pytest.org/en/latest/fixture.html
```

Comment:
More idiomatic would be to just use https://docs.pytest.org/en/latest/tmpdir.html

Reply:
Ok, will fix. Don't merge just yet.

Reply:
OK

Reply:
Fixed